BG/2DCode/raw/1.json

```diff
@@ -0,0 +1,26 @@
+    "BASE45": "NCFOXN%TSMAHN-HRQ8 QAN7K*J9%AL0II/M5-36HD7-TMQZ4A+F5DOC.I PFNFOQHILC90GBYPLR-SCG1CSQ6U7SSQY%S5L51I0N:4UO5920R2EX:6N5FQ7EUN6/DKC3277I$*S*FKC KXU7C KPLI8J4RK46YB8L46P8N$KKQS7DS2*N.SSBNKA.G.P6A8IM%OVNI*$K3$OHBW24FAL86H0YQCVLB*$K8KG+9RR$F+ F%J00N89M40%KLR2A KZ*U0I1-I0*OC6H0/VMNPM/UESJ0A5LCJ0PIA/$OHAE KMVEG0V1JZI5OI9YI:8DG:D%PD5DLAWC$ZJ*DJWP42W5KDT%%VANAKT5D86$RO%K1BPIFXU1YKIP5IJEYVBHQE73K4-G988*DF7Z23LT%882%K8%MQOIZ40M32QEWM.SY$NM*NR22C3CBL3M3BSSKI%B4PIQJAZGA2:UG%UYE47S17AJVC3%+0I7WF9D$6AK3UT E+YT/SFM0KK+EVVV5HCJRIK T IHBRI%ZKEJ9%6Q FVDHV %5% 92 B+VQN.J$M7T6JL2MK/1P/UU6ERGLLJR:T39UCC$N",
+    "PREFIX": "HC1:NCFOXN%TSMAHN-HRQ8 QAN7K*J9%AL0II/M5-36HD7-TMQZ4A+F5DOC.I PFNFOQHILC90GBYPLR-SCG1CSQ6U7SSQY%S5L51I0N:4UO5920R2EX:6N5FQ7EUN6/DKC3277I$*S*FKC KXU7C KPLI8J4RK46YB8L46P8N$KKQS7DS2*N.SSBNKA.G.P6A8IM%OVNI*$K3$OHBW24FAL86H0YQCVLB*$K8KG+9RR$F+ F%J00N89M40%KLR2A KZ*U0I1-I0*OC6H0/VMNPM/UESJ0A5LCJ0PIA/$OHAE KMVEG0V1JZI5OI9YI:8DG:D%PD5DLAWC$ZJ*DJWP42W5KDT%%VANAKT5D86$RO%K1BPIFXU1YKIP5IJEYVBHQE73K4-G988*DF7Z23LT%882%K8%MQOIZ40M32QEWM.SY$NM*NR22C3CBL3M3BSSKI%B4PIQJAZGA2:UG%UYE47S17AJVC3%+0I7WF9D$6AK3UT E+YT/SFM0KK+EVVV5HCJRIK T IHBRI%ZKEJ9%6Q FVDHV %5% 92 B+VQN.J$M7T6JL2MK/1P/UU6ERGLLJR:T39UCC$N",
+    "TESTCTX": {
+        "VERSION": 1,
+        "JWK": {
+            "kty": "EC",
+            "crv": "P-256",
+            "x": "f-dNj6CM9DdYbp-xsj2UTykKM-kfcGM5AAdGjE2P-gE",
+            "y": "7fIMKhi9L7ba7iNq7HcBAuWCCthKAPsWkCPylXuVPUY",
+            "d": "7aMNi-kJWZhcPOtvgrK-mt42VdWYJ3HGhjOcsWSXEzI",
+            "x5c": [
+              "MIIBaDCCAQ4CEQCcLypPcNRacesaaUFGxQadMAoGCCqGSM49BAMCMDYxJzAlBgNVBAMMHk5hdGlvbmFsIFRFU1QgQ1NDQSBvZiBCdWxnYXJpYTELMAkGA1UEBhMCQkcwHhcNMjEwNDIwMTE0MzQ1WhcNMjYwMzA1MTE0MzQ1WjA6MSswKQYDVQQDDCJURVNUIERTQyBudW1iZXIgd29ya2VyIG9mIEJ1bGdhcmlhMQswCQYDVQQGEwJCRzBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABH/nTY+gjPQ3WG6fsbI9lE8pCjPpH3BjOQAHRoxNj/oB7fIMKhi9L7ba7iNq7HcBAuWCCthKAPsWkCPylXuVPUYwCgYIKoZIzj0EAwIDSAAwRQIgKcW3IItuUk9dldKDcE9ybZMTb4JZXM2n4w1V63w49gACIQDI3H8tDaSnl7nFaAbUpoE9g0koKL9AfQg9xljyezL/Ew=="
+            ]
+        },
```

Comment:
Unfortunately I could not generate x5c certificate with both keys, so I included private key in JWK

Reply:
> @stamo Looks very good! We still have two open discussion points about the time format and the versioning. JSON Timestamp was proposed for validation clock and semver for the versioning. If this is closed, i merge your test data. You can join the discussion here: #2
> 
> The JWK private key part is as well a good hint. Thanks.

What? The private key is of no interest than the signer, so let's be clear about the tests. 

> @stamo Looks very good! We still have two open discussion points about the time format and the versioning. JSON Timestamp was proposed for validation clock and semver for the versioning. If this is closed, i merge your test data. You can join the discussion here: #2
> 
> The JWK private key part is as well a good hint. Thanks.

Why is the private key needed? The blob has been signed by the country that provides the testdata, and only the certificate/public key is needed to verify.

Reply:
Yes you are right, but take a look at the test procedure. The idea is to validate every step. So, one of the steps is generation of COSE object and it would be impossible without private key.  If we only have to validate barcode we definately do not need private key

Reply:
But the test instruction: "Sign with the JWK and compare the result to the COSE content." is a bit strange.

Clearly they must mean "validate the signature of the COSE content using the certificate from the JWK"?

Reply:
It was strange for me too, but decided to provide the key.

[https://github.com/eu-digital-green-certificates/dgc-testdata#jwk-content](url)

there you can see: **"The key pair to sign and validate the data structures. This is defined as x5c datastructure which contains the complete key pair"**

May be @SchulzeStTSI can clarify it

Reply:
The idea was to provide a full set of test data which allows others to debug step by step the outcomes. To insert the private (test) key, could be helpful. But if you reject it, we can remove it from the spec. I have second pull request open, where we can remove it.

Reply:
At least make it optional. Also see my issue at https://github.com/eu-digital-green-certificates/dgc-testdata/issues/5.